RecoParticleFlow/PFProducer/plugins/kdtrees/KDTreeLinkerTrackHcal.cc

```diff
@@ -206,142 +159,78 @@ void KDTreeLinkerTrackHcal::searchLinks() {
     float rangephi = (cristalPhiEtaMaxSize_ * (1.5 + 0.5) + 0.2 * fabs(dHphi)) * inflation;
 
     // We search for all candidate recHits, ie all recHits contained in the maximal size envelope.
-    std::vector<reco::PFRecHit const*> recHits;
+    std::vector<size_t> recHits;
     KDTreeBox trackBox(tracketa - rangeeta, tracketa + rangeeta, trackphi - rangephi, trackphi + rangephi);
     tree_.search(trackBox, recHits);
+    LogDebug("KDTreeLinkerTrackHcal") << "found " << recHits.size() << " associated to track " << itrack;
 
     // Here we check all rechit candidates using the non-approximated method.
-    for (auto const& recHit : recHits) {
-      const auto& rhrep = recHit->positionREP();
-      const auto& corners = recHit->getCornersREP();
-
-      double rhsizeeta = fabs(corners[3].eta() - corners[1].eta());
-      double rhsizephi = fabs(corners[3].phi() - corners[1].phi());
+    for (const size_t irechit : recHits) {
+      double rhsizeeta =
+          fabs(rechitTable.get<pf::rechit::Corner3eta>(irechit) - rechitTable.get<pf::rechit::Corner1eta>(irechit));
```

Comment:
some more compact wrap like `delta<Corner3eta, Corner1eta>(rechitTable, irechit)` (or smth similar) may be useful